python/adbc_driver_manager/adbc_driver_manager/dbapi.py

```diff
@@ -372,6 +378,11 @@ def close(self) -> None:
         if self._closed:
             return
 
+        # Close all open cursors first to avoid RuntimeError from
+        # AdbcConnection about open children
+        for cursor in list(self._cursors):
```

Comment:
nit: can't you iterate the WeakSet directly?

It appears the `list` guard is well out of date: https://stackoverflow.com/questions/12428026/safely-iterating-over-weakkeydictionary-and-weakvaluedictionary